log4j-api-test/src/test/java/org/apache/logging/log4j/message/ParameterFormatterTest.java

```diff
@@ -68,15 +67,12 @@ void test_pattern_analysis(
         }
     }
 
-    @ParameterizedTest
-    @CsvSource({"1,foo {}", "2,bar {}{}"})
-    void format_should_fail_on_insufficient_args(final int placeholderCount, final String pattern) {
-        final int argCount = placeholderCount - 1;
-        assertThatThrownBy(() -> ParameterFormatter.format(pattern, new Object[argCount], argCount))
-                .isInstanceOf(IllegalArgumentException.class)
-                .hasMessage(
-                        "found %d argument placeholders, but provided %d for pattern `%s`",
-                        placeholderCount, argCount, pattern);
```

Comment:
While we don't provide an equivalent `assertThatLoggedBy` method to test that a certain status logger message was issued (we probably should), you can add an `org.apache.logging.log4j.Logger statusLogger` parameter to `ParameterFormatter.format` and detect status messages as in [`ProviderUtilTest`](https://github.com/apache/logging-log4j2/blob/ed4c72e679951fe78fe4121436085444bc87b388/log4j-api-test/src/test/java/org/apache/logging/log4j/util/ProviderUtilTest.java).

Reply:
No, this is not needed. This file was already testing logged `StatusLogger` messages – scroll down, you will see the associated @SeasonPanPan deleted. They need to simply revert those changes and adapt them accordingly.

Reply:
Yes, I find the codes to check status logger what I deleted. I am adding the test cases.